### PR TITLE
[WIN32SS][NTUSER] Fix caret position of Ghostscript

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fix caret position of Ghostscript.
JIRA issue: [CORE-13934](https://jira.reactos.org/browse/CORE-13934), [CORE-15656](https://jira.reactos.org/browse/CORE-15656).

### CODE REVIEW

**Comment by katahiromz** [comment]
BEFORE:
gs-before
AFTER:
gs-after

**Comment by JoachimHenze** [comment]
This commit fixed also CORE-12240. Good!

**Comment by JoachimHenze** [comment]
This commit also (fixed or at least massively improved) CORE-10201 and CORE-13603. Good.

File: win32ss/user/ntuser/caret.c [hunk 1]
```diff
@@ -41,6 +43,13 @@ co_IntDrawCaret(PWND pWnd, PTHRDCARETINFO CaretInfo)
        NtGdiSaveDC(hdc);
     }
 
+    IntGetClientRect(pWnd, &rcClient);
+    NtGdiIntersectClipRect(hdc,
```
Comment (tkreuzer) [inline win32ss/user/ntuser/caret.c]: There is still no explanation, why this is required. The returned DC is an owned/class DC (as opposed to a cache DC), i.e. it is persistent and should never change under the hood. This code changes the DC's clipping region. Instead my suggestion is to use DCX_CACHE, which gives a fresh DC from the cache and no intersect and no save/restore should be required.

**Pull request merged**